test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1167,46 +1170,58 @@ public async Task GetBytes()
         }
 
         [Test]
-        public async Task GetStream([Values(true, false)] bool isAsync)
+        public async Task GetStreamSecondTimeThrows([Values(true, false)] bool isAsync)
         {
+            var expected = new byte[] { 1, 2, 3, 4, 5, 6, 7, 8 };
             var streamGetter = BuildStreamGetter(isAsync);
 
-            using (var conn = await OpenConnectionAsync())
+            using var conn = await OpenConnectionAsync();
+            using var cmd = new NpgsqlCommand($"SELECT {EncodeByteaHex(expected)}::bytea", conn);
+            using var reader = await cmd.ExecuteReaderAsync(Behavior);
+
+            await reader.ReadAsync();
+
+            using var stream = await streamGetter(reader, 0);
+
+            Assert.That(async () => await streamGetter(reader, 0),
+                Throws.Exception.TypeOf<InvalidOperationException>());
+        }
+
+        [Test]
+        public async Task GetStream([Values(true, false)] bool isAsync)
+        {
+            var binary = MemoryMarshal
```

Comment:
Just to make sure it's intentional: this doesn't create an array of 1024 consecutive byte values, but rather an array of 4096 byte values which represent 4-byte int data... Is there any reason? It's usually easier when debugging tests to have simple consecutive data, in order to understand what's going on.

Reply:
Previously there was a comment containing a to-do. The idea was to test sequential mode truly, but for this we should has more data than the buffer could contain. So multiply 4096 by 2 and add some extra from the PostgreSQL protocol.